tests/test_ssdp.py

```diff
@@ -0,0 +1,190 @@
+import queue
```

Comment:
Tests not closely looked at for this PR yet either.  I do love that you're adding tests for all of the newly added code.